### DISCUSSION TITLE
After updating to Thunderbird 102.0 the addon stopped working / crash TB

### DISCUSSION THREAD

**Discussion by JBaae** [kewisch/gdata-provider#462] [Support]
Hi All,

after updating my Thunderbird to 102.0 it crashes/hung-up during start. I had to disable the Google-Calender-Provider addon to have the Thunderbird working.

Any ideas what to do?

All the best

Jens

**Comment by Smurfengrunk** [comment]
Same problem here. The add-on makes Thunderbird completely unresponsive, but still hogs quite some CPU (about 30-40%).
Disabling the provider solves the problem. I'm now using the built-in calendar, which works pretty well but can't add subscribed calendars - only my own are showing.

**Reply by wsmwk** [reply]
@JBaae, @Smurfengrunk 

Are you still seeing this problem when using a newer version of Thunderbird and a newer version of provider?

**Reply by Smurfengrunk** [reply]
Sadly yes, using Thunderbird 102.1.2 and addon 102.0.0 (august 18 2022).

**Reply by Smurfengrunk** [reply]
After deleting all calendars, removing and adding the extension it did work partially.
Console messages:
"14:58:10,906 1661345890906	addons.xpi	WARN	Checking C:\Program Files\Mozilla Thunderbird\distribution\extensions for addons
14:58:11,279 Den här sidan har bristfällig kod. Sidlayouten kan påverkas. För standardläge använd "<!DOCTYPE html>". hiddenWindow.html
14:58:12,043 Trying to load C:\Program Files\Mozilla Thunderbird\libotr.dll OTRLib.jsm:64:11
14:58:12,046 Successfully loaded OTR library C:\Program Files\Mozilla Thunderbird\libotr.dll OTRLib.jsm:72:13
14:58:12,063 Successfully loaded OpenPGP library rnp.dll version 0.16+git20220124.f06439f7.MZLA from C:\Program Files\Mozilla Thunderbird\rnp.dll RNPLib.jsm:100:15
14:58:12,075 Found 0 public keys and 0 secret keys (0 protected, 0 unprotected) RNPLib.jsm:301:15
14:58:12,101 services.settings: Failed to load last_modified.json: TypeError: NetworkError when attempting to fetch resource. Utils.jsm:330
14:58:12,828 No chrome package registered for chrome://sendtocategory/content/category_tools.js 2
14:58:15,174 tb.account.size_on_disk - Trying to set an unsigned scalar to a negative number. 4
14:58:17,129 Försök att ställa in en förbjuden header nekades: Content-Length 2 cardbookWebDAV.js:386:17
14:58:54,007 1661345934007	addons.xpi	WARN	Exception running bootstrap method shutdown on {a62ef8ec-5fdc-40c2-873c-223b8a6925cc}: TypeError: this.extension is undefined(resource://gre/modules/Extension.jsm:2149:18) JS Stack trace: shutdown@Extension.jsm:2149:18
callBootstrapMethod@XPIProvider.jsm:1848:33
_shutdown@XPIProvider.jsm:1981:17
14:59:00,435 Denna sida använder icke-standardegenskapen “zoom”. Överväg att använda calc() i relevanta egenskapsvärden eller använd “transform” tillsammans med “transform-origin: 0 0”. search
14:59:00,518 Vissa kakor missbrukar det rekommenderade attributet "SameSite" 15
14:59:00,613 downloadable font: kern: Too large subtable (font-family: "Open Sans" style:normal weight:700 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Bold.ttf?5014568
14:59:00,613 downloadable font: Table discarded (font-family: "Open Sans" style:normal weight:700 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Bold.ttf?5014568
14:59:00,613 downloadable font: kern: Too large subtable (font-family: "Open Sans" style:italic weight:700 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-BoldItalic.ttf?78b08a6
14:59:00,613 downloadable font: Table discarded (font-family: "Open Sans" style:italic weight:700 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-BoldItalic.ttf?78b08a6
14:59:00,613 downloadable font: kern: Too large subtable (font-family: "Open Sans" style:normal weight:400 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Regular.ttf?629a55a
14:59:00,613 downloadable font: Table discarded (font-family: "Open Sans" style:normal weight:400 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Regular.ttf?629a55a
14:59:00,630 downloadable font: kern: Too large subtable (font-family: "Open Sans" style:italic weight:400 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Italic.ttf?c7dcce0
14:59:00,630 downloadable font: Table discarded (font-family: "Open Sans" style:italic weight:400 stretch:100 src index:2) source: https://addons.thunderbird.net/static/css/impala/fonts/OpenSans-Italic.ttf?c7dcce0
14:59:07,105 InstallTrigger är föråldrad och kommer att tas bort i framtiden. impala-min.js:1:34833
14:59:07,105 InstallTrigger.install() är föråldrad och kommer att tas bort i framtiden. För mer hjälp https://extensionworkshop.com/documentation/publish/self-distribution/ impala-min.js:1:34848
15:00:11,129 Failed to get the active browserId while starting the profiler. recording-utils.js:53:11
15:00:26,158 Calendar: [calStorageCalendar] Message: executeSyncItemStatement exception
Connection Ready: true
Last DB Error Number: 6
Last DB Error Message: database table is locked
Database File: C:\Users\Tomas\AppData\Roaming\Thunderbird\Profiles\hp766xdu.default-release\calendar-data\cache.sqlite
Last DB Statement: [object StatementJSHelper]
Last Statement param [0]: undefined
Last Statement param [1]: undefined
Last Statement param [2]: undefined
Last Statement param [offline_journal]: undefined
Last Statement param [id]: undefined
Last Statement param [cal_id]: undefined
Last Statement param [length]: 3
Exception: [Exception... "Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep]"  nsresult: "0x8052000e (NS_ERROR_FILE_IS_LOCKED)"  location: "JS frame :: resource:///modules/calendar/CalStorageDatabase.jsm :: executeSyncItemStatement :: line 131"  data: no]
1: [resource:///modules/calendar/CalStorageDatabase.jsm:263] logError
2: [resource:///modules/calendar/CalStorageDatabase.jsm:133] executeSyncItemStatement
3: [resource:///modules/calendar/CalStorageMetaDataModel.jsm:46] deleteMetaDataById
4: [resource:///modules/CalStorageCalendar.jsm:519] setMetaData
5: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:88] saveItemMetadata
6: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:1232] commitItem
CalStorageDatabase.jsm:263
15:00:26,158 NS_ERROR_FILE_IS_LOCKED: Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep] CalStorageDatabase.jsm:131
15:00:26,193 uncaught exception: [calCachedCalendar] replay action failed: <unknown>, uri=googleapi://tomas.n.lundqvist@gmail.com/?calendar=lundqvist.agneta%40gmail.com, result=Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep], operation=[xpconnect wrapped calIOperation] calCachedCalendar.js:77:23
15:00:27,964 Calendar: [calStorageCalendar] Message: executeSyncItemStatement exception
Connection Ready: true
Last DB Error Number: 6
Last DB Error Message: database table is locked
Database File: C:\Users\Tomas\AppData\Roaming\Thunderbird\Profiles\hp766xdu.default-release\calendar-data\cache.sqlite
Last DB Statement: [object StatementJSHelper]
Last Statement param [0]: undefined
Last Statement param [1]: undefined
Last Statement param [2]: undefined
Last Statement param [offline_journal]: undefined
Last Statement param [id]: undefined
Last Statement param [cal_id]: undefined
Last Statement param [length]: 3
Exception: [Exception... "Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep]"  nsresult: "0x8052000e (NS_ERROR_FILE_IS_LOCKED)"  location: "JS frame :: resource:///modules/calendar/CalStorageDatabase.jsm :: executeSyncItemStatement :: line 131"  data: no]
1: [resource:///modules/calendar/CalStorageDatabase.jsm:263] logError
2: [resource:///modules/calendar/CalStorageDatabase.jsm:133] executeSyncItemStatement
3: [resource:///modules/calendar/CalStorageMetaDataModel.jsm:46] deleteMetaDataById
4: [resource:///modules/CalStorageCalendar.jsm:519] setMetaData
5: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:88] saveItemMetadata
6: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:1232] commitItem
CalStorageDatabase.jsm:263
15:00:27,964 NS_ERROR_FILE_IS_LOCKED: Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep] CalStorageDatabase.jsm:131
15:00:27,990 uncaught exception: [calCachedCalendar] replay action failed: <unknown>, uri=googleapi://tomas.n.lundqvist@gmail.com/?calendar=addressbook%23contacts%40group.v.calendar.google.com, result=Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep], operation=[xpconnect wrapped calIOperation] calCachedCalendar.js:77:23
15:00:28,677 Calendar: [calStorageCalendar] Message: executeSyncItemStatement exception
Connection Ready: true
Last DB Error Number: 6
Last DB Error Message: database table is locked
Database File: C:\Users\Tomas\AppData\Roaming\Thunderbird\Profiles\hp766xdu.default-release\calendar-data\cache.sqlite
Last DB Statement: [object StatementJSHelper]
Last Statement param [0]: undefined
Last Statement param [1]: undefined
Last Statement param [2]: undefined
Last Statement param [offline_journal]: undefined
Last Statement param [id]: undefined
Last Statement param [cal_id]: undefined
Last Statement param [length]: 3
Exception: [Exception... "Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep]"  nsresult: "0x8052000e (NS_ERROR_FILE_IS_LOCKED)"  location: "JS frame :: resource:///modules/calendar/CalStorageDatabase.jsm :: executeSyncItemStatement :: line 131"  data: no]
1: [resource:///modules/calendar/CalStorageDatabase.jsm:263] logError
2: [resource:///modules/calendar/CalStorageDatabase.jsm:133] executeSyncItemStatement
3: [resource:///modules/calendar/CalStorageMetaDataModel.jsm:46] deleteMetaDataById
4: [resource:///modules/CalStorageCalendar.jsm:519] setMetaData
5: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:88] saveItemMetadata
6: [resource://gdata-provider/legacy/modules/gdataUtils.jsm:1232] commitItem
CalStorageDatabase.jsm:263
15:00:28,677 NS_ERROR_FILE_IS_LOCKED: Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep] CalStorageDatabase.jsm:131
15:00:28,692 uncaught exception: [calCachedCalendar] replay action failed: <unknown>, uri=googleapi://tomas.n.lundqvist@gmail.com/?calendar=pingstkyrkanelim%40gmail.com, result=Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [mozIStorageStatement.executeStep], operation=[xpconnect wrapped calIOperation] calCachedCalendar.js:77:23
15:01:52,042 Den här sidan har bristfällig kod. Sidlayouten kan påverkas. För standardläge använd "<!DOCTYPE html>". {e3b33b99-0289-4526-80d4-0713809e6f12}
15:01:52,042 Den här sidan har bristfällig kod. Sidlayouten kan påverkas. För standardläge använd "<!DOCTYPE html>". blank
15:01:52,050 Den här sidan har bristfällig kod. Sidlayouten kan påverkas. För standardläge använd "<!DOCTYPE html>". INBOX>17338
15:01:52,392 Vissa kakor missbrukar det rekommenderade attributet "SameSite" 2"

Performance profile:
https://share.firefox.dev/3PMCFi7

**Reply by Smurfengrunk** [reply]
...and after less than a minute of synchronization the whole package freezes indefinitely again. Trying to get a performance log when starting up doesn't work.
Updated TB to 102.2.0 but that doesn't change anything sadly.

**Comment by Gothico74** [comment]
Addon version is 102.0.0.
I updated TB to 102.2.0 x64 today.
And i have to say, yes sadly this great addon causes TB to freeze at startup for one - two minutes till i am able to use TB.
Disabled - i am not able to edit and sync my calendars (gmail).

**Reply by Gothico74** [reply]
i updated the plugin to 102.0.1
same issue
first login gave me an time-out, so somethings wronge with the authorization prozess

**Reply by oink1893** [reply]
similar issue...just that it hangs much longer than 2 min on startup

**Comment by wsmwk** [comment]
Please obtain a performance profile as documented at https://support.mozilla.org/en-US/kb/profiling-thunderbird-performance

**Reply by Gothico74** [reply]
Error: DataCloneError: JSWindowActorParent.sendAsyncMessage: JSWindowActorParent serialization error: data too large, in actor 'WebChannel'

upsi

**Comment by Gothico74** [comment]
So what i did was:

disabling the addon and cancel every calendar i use to sync
starting the addon again and nothing bad happens
adding calendars to sync: use google and add my email
coosing three of them, added
sync is in progress and stops (freezes) at 800 from 1000 entrys
two minute a frozen TB, and than finished all
thats the behaviour every time TB starts

but i have three errors in the console:

`GEThttps://www.googleapis.com/calendar/v3/users/me/calendarList/v...............7rif@import.calendar.google.com [HTTP/3 404 Not Found 90ms]`

`NS_ERROR_XPC_JSOBJECT_HAS_NO_FUNCTION_NAMED: JavaScript component does not have a method named: "getItems"'JavaScript component does not have a method named: "getItems"' when calling method: [calICalendar::getItems]`

`uncaught exception: [calCachedCalendar] replay action failed: <unknown>, uri=googleapi://yesbaby@gmail.com?calendar=v...........................7rif%40import.calendar.google.com, result=[object Object], operation=[object Object]`

my calendars are showing up and are up-to-date, so nothings wrong....

**Comment by Wurya123** [comment]
Unfortunately the calendar plugin for Thunderbird also freezes for me since the Thunderbird update to 102.2 on Windows 11 version 10.0.22000 build 22000.

I get the following error message:
 [Exception... "[JavaScript Error: "getSelectedCalendar() is null" {file: "chrome://calendar/content/calendar-task-editing.js" line: 152}]'[JavaScript Error: "getSelectedCalendar() is null" {file: "chrome://calendar/content/calendar-task-editing.js" line: 152}]' when calling method: [calIObserver::onPropertyChanged]"  nsresult: "0x80570021 (NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS)"  location: "JS frame :: resource:///modules/calendar/utils/calDataUtils.jsm :: notify :: line 38"  data: yes]

**Reply by Wurya123** [reply]
My issue is resolved since TB Version 102.3.2 . 

I had a look into the release notes and my assumption is that the fix relates to "Fix: Thunderbird would hang if calendar event exceeded the year 2035"

**Comment by Smurfengrunk** [comment]
For me it doesn't just hang for a minute or two, it freezes until I force close and restart it with the extension disabled. I've tried with deleting all calendars and adding them back again, and that makes it work for a short while but then the process starts all over again.

**Comment by wsmwk** [comment]
FWIW, calendar entries repeating appointments with expiration dates far into the future can cause performance issues.

**Comment by Soundmanbe** [comment]
Tried all possibilities, reloading all calendars... no way, freezing during the synchronisation. No other way than disable the add-on. No more access to the GMAIL Calendar. TB 102.3.1 (64-bit) Linux

**Comment by electrikweb** [comment]
I was about to write "Same here. ", as disabling the calendar provider plugin solved the issue. When I looked up my TB version, I saw there was an update pending (to 102.3.2 (32-bit)). After installing it, I no longer have to wait 5 mins. Everything loads smoothly and quickly again. Plugin works fine now

**Reply by Wurya123** [reply]
For me it was the same. Since 102.3.2 , 64-bit in my case, everything is okay.
